WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -702,13 +702,6 @@ extension MediaLibraryViewController: UIViewControllerRestoration {
     }
 }
 
-fileprivate extension Blog {
-    var userCanUploadMedia: Bool {
-        // Self-hosted non-Jetpack blogs have no capabilities, so we'll just assume that users can post media
-        return capabilities != nil ? isUploadingFilesAllowed() : true
-    }
-}
-
```

Comment:
:wave: @frosty ! I was wondering if you might have any thoughts on whether my decision to move this logic into `Blog+Capabilities` is sound. I did this to avoid duplicating the logic both here and with the `GutenbergViewController` since the `capabitlies != nil` part of the check seemed like a tricky little gotcha, but I wonder if there's a better way to do that. Or perhaps this small bit of duplication isn't worth avoiding. 

Adding something to `Blog+Capabilities` just felt like it might not be the right decision, and I'm hoping to get some input from someone who is more familiar with Swift and WPiOS than I am (admittedly, this is a low bar). You added this method, so that's why I pinged you, but It has been [almost 5 years](https://github.com/wordpress-mobile/WordPress-iOS/commit/a0940f02bada3309f83d297cb799b44aa4fddda7) since you did that, so I'm going to go out on a limb and guess that it might not be fresh in your mind. 🤣  Just let me know if you don't have time to take a look at this or if you think there might be another person who would be better to ping.  

If you'd like to see the refactor in isolation you can look at https://github.com/wordpress-mobile/WordPress-iOS/pull/18229/commits/af5a94a5a3f7eab42611be1fb6d924122d2d1f05.

Reply:
Hey, apologies for missing the ping! I think this seems like a good location for this code, as it fits in with the other capabilities checks. I appreciate the comment in there too to explain why we're counting nil `capabilities` as `true`.

It bothers me slightly that the name `userCanUploadMedia` doesn't match the format of the other methods here (`isXAllowed`), but I can't really think of a better name that does match! So I think it's fine. Could we add a `///` doc comment to the method, like the others in here?

Thanks for checking!